alpenhorn/acquisition.py

```diff
@@ -25,6 +32,114 @@ class AcqType(base_model):
+            AcqInfoBase describing the type of acquisition.
+        """
+
+        try:
+            cls.get(name=acq_info._acq_type)
+        except pw.DoesNotExist:
+            log.info("Create AcqType entry for \"%s\"" % acq_info._acq_type)
+            cls.create(name=acq_info._acq_type)
+
+        # Add to registry
+        cls._registered_acq_types[acq_info._acq_type] = acq_info
+
+    @classmethod
+    def check_registration(cls):
+        """Check that all AcqTypes known to the database have a
+        registered handler.
+        """
+
+        # Get the list of types names from the database
+        db_acq_types = [row[0] for row in cls.select(cls.name).tuples()]
+
+        # Get the names of all the registered types
+        reg_acq_types = cls._registered_acq_types.keys()
+
+        # Find any missing types
+        missing_acq_types = set(db_acq_types) - set(reg_acq_types)
+
+        if len(missing_acq_types):
+            raise RuntimeError('AcqTypes %s have no registered handler.' %
+                               repr(missing_acq_types))
+
+    def is_type(self, acqname, node):
+        """Does this acqisition type understand this directory?
+
+        Parameters
+        ----------
+        acqname : string
+            Name of the acquisition we are checking.
+        node : StorageNode
+            The node we are importing from. Needed so we can inspect the actual
+            acquisition.
+
+        Returns
+        -------
+        is_type : boolean
+        """
+        return self.acq_info._is_type(acqname, node.root)
+
+    @classmethod
+    def detect(cls, acqname, node):
+        """Try and find an acquisition type that understands this directory.
+
+        Parameters
+        ----------
+        acqname : string
+            Name of the acquisition we are trying to find the type of.
+        node : StorageNode
+            The node we are importing from. Needed so we can inspect the actual
+            acquisition.
+        """
+
+        # Iterate over all known acquisition types to try and find one that matches
+        # the directory being processed
+        for acq_type in cls.select():
+
+            if acq_type.is_type(acqname, node):
+                return acq_type
+
+        return None
+
+    @property
+    def acq_info(self):
+        """The AcqInfo table for this AcqType.
+        """
+        return self.__class__._registered_acq_types[self.name]
+
+    @property
+    def file_types(self):
+        """The FileTypes supported by this AcqType.
+        """
+
+        def _resolve(x):
+            if isinstance(x, str):
+                return x
+            elif isinstance(x, type) and issubclass(x, FileInfoBase):
+                return x.file_type
+            else:
+                raise RuntimeError("FileType %s not understood." % repr(x))
+
+        # Resolve the file_types list of the AcqInfo to the names
+        file_type_names = [_resolve(x) for x in self.acq_info._file_types]
```

Comment:
`_file_types` has the option of being a string such that it can easily be set from a config file. In theory the resolve step could be done as the configuration is set, but I figured it was easy enough to do it here. As for why it can also take an instance, that seems like it could be useful and I figured why not?